### ISSUE TITLE
Integrate FixedIntervalNumberObserver and NumberObserver

### ISSUE THREAD

**Issue by kaizu** [ecell/ecell4#237]
`NumberObserver` could be the special case of `FixedIntervalNumberObserver` with `dt` equal to `0`.

```python
obs1 = NumberObserver(1.0)  # FixedIntervalNumberObserver
obs2 = NumberObserver(0.0)  # NumberObserver
```

The main objective of this proposal is to simplify the name of objects often used.

**Comment by kaizu** [comment]
`TimingNumberObserver` is also available: `obs3 = NumberObserver([0.0, 3.0, 5.0])`.

It might be enough to provide an interface to make a proper observer:

```python
obs1 = create_number_observer(1.0)  # -> FixedIntervalNumberObserver
obs2 = create_number_observer(0.0)  # -> NumberObserver
obs3 = create_number_observer([0.0, 3.0, 5.0])  # -> TimingNumberObserver
```

But `create_number_observer` is too long to type often.

**Comment by kaizu** [comment]
Use `number_observer`:

```python
obs1 = number_observer(1.0)  # -> FixedIntervalNumberObserver
obs2 = number_observer()  # -> NumberObserver
obs3 = number_observer([0.0, 3.0, 5.0])  # -> TimingNumberObserver
```
